hw_9_corr.py

```diff
@@ -0,0 +1,287 @@
+
+    def add_material(self, name, details, quantity, availability, price):
+        material = Material(name, details, quantity, availability, price)
+        self.stock.append(material)
+
+    def remove_material(self, material):
+        material = self.find_material(material)
+        if material:
+            self.stock.remove(material)
+
+    def current_quantity(self, material):
+        material = self.find_material(material)
+        if material:
+            print('Current stock of {0} is {1}'.format(material.name, material.quantity))
+        else:
+            print('Material does not exist.')
+
+    def current_quantity_all(self):
+        if self.stock:
+            for material in self.stock:
+                self.current_quantity(material.name)
+        else:
+            print('Warehouse is empty.')
+
+
+class Category(Warehouse):
+    def __init__(self, name):
+        self.name = name
+        self.members = []
+        super().__init__()
+
+    def add_to_category(self, material):
+        if isinstance(material, Material):
+            self.members.append(material)
+            material.category = self.name
+        else:
+            print('Material does not belong to material class.')
+
+    def print_category(self):
+        id = 0
+        if self.members:
+            for material in self.members:
+                id += 1
+                print(id, 'Material name: ', material.name, 'Material quantity :', material.quantity)
+
+
+class Basket():
+    def __init__(self):
+        self.content = {}
+        self.total = 0
+
+    def add_to_basket(self, material, value):
+        if isinstance(material, Material):
+            if material.quantity >= value:
+                self.content[material] = value
+                self.total += value * material.price
+            else:
+                raise ValueError('Material quantity in Warehouse is not enough to process your order.')
+        else:
+            raise TypeError('Material has to be an instance of class Material.')
+
+    def print_basket(self):
+        id = 0
+        if self.content:
+            print('Basket')
+            for material in self.content:
+                id += 1
+                print(id, 'Material: ', material.name, 'Quantity: ', self.content[material], 'Price: ', material.price,
+                      'Total: ', self.content[material] *  material.price)
+            print('Grand Total: ', self.total)
+
+
+class Order(Basket):
+    def __init__(self):
+        self.items = {}
+        global order_id
+        order_id += 1
+        self.order_id = order_id
+        self.date_purchased = datetime.datetime.today()
+        super().__init__()
+
+    def print_order(self):
+        id = 0
+        print('Order ID: ', self.order_id)
+        print('Date purchased: ', self.date_purchased)
+        for material in self.items:
+            id += 1
+            print(id, 'Material: ', material.name, 'Quantity: ', self.items[material])
+
+
+order_id = 0
```

Comment:
why order_id as a GLOBAL? this var should be only for Order class

Reply:
var order_id has been moved to the Order class and set as a class atribute,